pdns/syncres.cc

```diff
@@ -1111,6 +1127,9 @@ void SyncRes::getBestNSFromCache(const DNSName &qname, const QType& qtype, vecto
   bestns.clear();
   bool brokeloop;
   do {
+    if (!subdomain.isPartOf(cutOffDomain)) {
```

Comment:
I did not test it but at first sight it looks like we will look the NS for `cutOffDomain` itself. We want that when it's `g_rootdnsname` but I'm not sure we do when it's `fwdomain`?

Reply:
Correct, but I did not want to complicate matters further. It should not hurt (TM).

Reply:
I wonder if we can somehow end up with `NS` records in the cache for a forwarded domain? We should refuse them from other authoritative servers than the ones we forward to, but what happens if these send us a `NS` record that does not match our forward?

Reply:
OK, I made the code more robust against that by using boost::optional and only doing cutoff when we are passed a cutoff point. This has the nice consequence that the original behaviour (with no cutoff) is not modified at all.